File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.Shape;
/* 
 * Renderer Constructor now takes in argument of type
 * Shape instead of Rectange, shape variable set to private
 */
public class Renderer {
    private Shape shape;
    
    Renderer(Shape shape) {
        this.shape = shape;
    }

    void draw() {
        double area = shape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}

File: typescript/src/renderer.ts
import { Shape } from "./shapes/shape"

/* 
 * newRenderer now takes in argument of type
 * Shape instead of Rectange
 */
function newRenderer(shape : Shape) {
    return {
        draw() {
            const area: number = shape.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { newRenderer }